src/components/Repay/TokenUserInput.tsx

```diff
@@ -0,0 +1,54 @@
+import Image from 'next/image'
+import { Token } from '../Tokens'
+
+const style = {
+  tokenUserInputContainer: 'flex justify-around mt-2',
+  tokenAmountContainer: '',
+  loanAmountInput: 'bg-transparent placeholder:text-neutrals-600 outline-none w-full text-2xl text-neutrals-200',
+  maxButton: 'w-[2rem] text-center bg-neutrals-900 bg-opacity-60 drop-shadow-lg cursor-pointer hover:bg-opacity-70 text-xxs font-normal p-0.5 rounded-sm text-accents-royalBlue text-opacity-50 hover:text-opacity-80 mt-1',
+  tokenPairContainer: 'w-[12rem]',
+  tokenPairContent: 'flex justify-end space-x-3',
+  tokenPairIcons: 'relative w-[2rem] h-[2rem] self-center',
+  tokenAIcon: 'mt-0.5',
+  tokenBIcon: 'absolute top-0.5 right-0 -z-10',
+  tokenPairSymbol: 'text-lg',
+  tokenPairBalance: 'font-semibold text-xxs text-neutrals-600 tracking-wide text-right mt-2',
+}
+
+interface TokenUserInputProps {
+  token0: Token
+  token1: Token
+}
+
+export const TokenUserInput = ({ token0, token1 }: TokenUserInputProps) => {
```

Comment:
We don't want to have the main input to be the token input and the percentage on the side as discussed?

Reply:
What would it look like for reserve tokens then? In that case, you would need two sliders? Or will the amount you input take into account both tokens and do the split calculation?

Reply:
I think there's no LP/Reserve switch for this screen. The units of the big input should always represent how much LP tokens the user wants to repay. Thet total loan in LP tokens can be found using the loan() query of the smart contract. The pool currently only holds reserve tokens as collateral which the pool will swap for LP tokens when the user confirms repayment. It should be possible to calculate how much reserve tokens will be left over after repaying any amount of LP tokens. 

It is possible that the user's collateral cannot repurchase enough LP tokens to repay 100% if their collateral ratio is really out of balance because uniswap requires ratio of 50/50 to be traded for LP tokens. In that case the user needs to rebalance his collateral reserve tokens and then he can repay 100% of the loan. I think for MVP we can assume he can always 100% without rebalancing. If it's enough enough then let the contracts fail with error.

Reply:
good insight, thanks